qiita_db/metadata_template/base_metadata_template.py

```diff
@@ -1516,3 +1527,17 @@ def validate(self, restriction_dict):
                 "columns:\n\t%s.\nSee the Templates tutorial for a description"
                 " of these fields." % ";\n\t".join(warning_msg),
                 qdb.exceptions.QiitaDBWarning)
+
+    def unique_columns(self):
```

Comment:
This is gonna be extremely slow for large info files. Examples from the real system:
```python
from qiita_db.metadata_template.sample_template import SampleTemplate

def unique_columns(st):
    df = st.to_dataframe()
    n = len(df)
    return {k for k, v in df.nunique().iteritems() if v == n}

In [22]: %timeit unique_columns(SampleTemplate(550)) # moving pics
1 loop, best of 3: 662 ms per loop

In [23]: %timeit unique_columns(SampleTemplate(10317)) # AGP
1 loop, best of 3: 1min 16s per loop
```

As you can imagine something blocking for 1min 16s is not desirable and more in the sample info page which is pretty common and we normally expect fasts replies. 

Options:
- Do everything in DB - preferred but not sure if much faster
- Do not display only unique values but all columns - and option but don't love it
- Store unique values in redis and change every time the sample id is replaced as part of the update - kind of cool to store all summaries as part of the update 
- Other?